test/Microsoft.AspNetCore.Sockets.Tests/TestTaskExtensions.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Sockets.Tests
+{
+    public static class TestTaskExtensions
```

Comment:
This is something we want to do for all tests though. I think @mikeharder added something to our xunit test infrastructure to allow timing out tests.